src/main/scala/mesosphere/marathon/state/AppDefinition.scala

```diff
@@ -85,7 +86,8 @@ case class AppDefinition(
   def portIndicesAreValid(): Boolean = {
     val validPortIndices = 0 until hostPorts.size
     healthChecks.forall { hc =>
-      validPortIndices contains hc.portIndex
+      (hc.protocol == Protocol.COMMAND && hc.portIndex == 0) ||
```

Comment:
I guess the point is that the portIndex is actually ignored for COMMAND checks, so validating it could result in a confusing and unnecessary error.  That is, IMO we shouldn't tell the user to fix something in the request which doesn't actually have an effect on how the health check is conducted.